security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -168,18 +157,18 @@ func (sc *SecretCache) generateSecret(token string, t time.Time) (*SecretItem, e
 		return nil, err
 	}
 
-	return &SecretItem{
+	return &sds.SecretItem{
 		CertificateChain: certChainPER,
 		PrivateKey:       keyPEM,
-		token:            token,
-		lastUsedTime:     t,
-		createdTime:      t,
+		Token:            token,
+		LastUsedTime:     t,
+		CreatedTime:      t,
 	}, nil
 }
 
-func (sc *SecretCache) checkExpired(s *SecretItem) bool {
```

Comment:
Can you leave a TODO to state the criteria of expiration is TBD? Since we may want to respect the returned key cert TTL.

Also given we should refresh it before it becomes expired, suggest to change this to `shouldRefresh`.

Reply:
@quanjielin can you take a look on this? I think we need to reserve some grace period to make sure `now() <= TTL - envoy_reconnect_delay - CSR_round_trip delay`, that's why I suggest to rename it to shouldRefresh instead of isCertExpired.

Reply:
renamed and added the above comment in to TODO code comment, it's very accurate :) thanks